src/battle_script_commands.c

```diff
@@ -1267,6 +1267,20 @@ bool32 ProteanTryChangeType(u32 battler, u32 ability, u32 move, u32 moveType)
     return FALSE;
 }
 
+bool32 DoesSkyBattleCancelCurrentMove(u32 gCurrentMove)
+{
+    bool8 moveBannedFromSkyBattles = FALSE;
+
+    //if (gBattleMoves[gCurrentMove].flags & FLAG_DISABLED_IN_SKY_BATTLE)
+    if (gBattleMoves[gCurrentMove].skybattleBanned)
+        moveBannedFromSkyBattles = TRUE;
+
+    if ((gBattleStruct->rulesVariants.skyBattle) && (moveBannedFromSkyBattles)){
+        return TRUE;
+    }
+    return FALSE;
+}
+
```

Comment:
This function could be simplified to sth like

bool32 isMoveNotAllowedInSkyBattles(u32 move)
{
     return ((gBattleStruct->isSkyBattle && gBattleMoves[move].skybattleBanned);
}